automation/jenkins_build.sh

```diff
@@ -102,64 +147,88 @@ then
-		(cd $BUILD_DEPLOY_DIR && tar --remove-files  --use-compress-program pigz --directory=$DEPLOY_ARTIFACT -cvf ${DEPLOY_ARTIFACT}.tar.gz .)
+# Artifacts
+YOCTO_BUILD_DEPLOY="$WORKSPACE/build/tmp/deploy/images/$MACHINE"
+VERSION_HOSTOS=$(cat "$YOCTO_BUILD_DEPLOY/VERSION_HOSTOS")
+DEVICE_TYPE_JSON="$WORKSPACE/$MACHINE.json"
+SLUG=$(jq --raw-output '.slug' $DEVICE_TYPE_JSON)
+
+# Jenkins artifacts
+echo "[INFO] Starting creating jenkins artifacts..."
+deploy_build "$WORKSPACE/deploy-jenkins" "true"
+
+# Deploy
+if [ "$deploy" == "yes" ]; then
+    echo "[INFO] Starting deployment..."
+    if [ "$deployTo" == "production" ]; then
+        echo "[INFO] Pushing resinhup package to dockerhub and registry.resinstaging.io."
+        DOCKER_REPO="resin/resinos"
+        RESINREG_REPO="registry.resinstaging.io/resin/resinos"
+        # Make sure the tags are valid
+        # https://github.com/docker/docker/blob/master/vendor/github.com/docker/distribution/reference/regexp.go#L37
+        DOCKER_TAG="$(echo $VERSION_HOSTOS-$SLUG | sed 's/[^a-z0-9A-Z_.-]/_/g')"
+        RESINREG_TAG="$(echo $VERSION_HOSTOS-$SLUG | sed 's/[^a-z0-9A-Z_.-]/_/g')"
+        RESINHUP_PATH=$(readlink --canonicalize $WORKSPACE/build/tmp/deploy/images/$MACHINE/resin-image-$MACHINE.resinhup-tar)
+        if [ -f $RESINHUP_PATH ]; then
+            docker import $RESINHUP_PATH $DOCKER_REPO:$DOCKER_TAG
+            docker push $DOCKER_REPO:$DOCKER_TAG
+            docker rmi $DOCKER_REPO:$DOCKER_TAG # cleanup
+            docker import $RESINHUP_PATH $RESINREG_REPO:$RESINREG_TAG
+            docker push $RESINREG_REPO:$RESINREG_TAG
+            docker rmi $RESINREG_REPO:$RESINREG_TAG # cleanup
+        else
+            echo "[ERROR] The build didn't produce a resinhup package."
+            exit 1
+        fi
+    fi
+
+	# Deployment to s3
+	S3_VERSION_HOSTOS=$VERSION_HOSTOS
+	if [ "$DEVELOPMENT_IMAGE" == "yes" ]; then
+		S3_VERSION_HOSTOS=$VERSION_HOSTOS.dev
+	fi
+	S3_DEPLOY_DIR="$WORKSPACE/deploy-s3"
+	S3_DEPLOY_IMAGES_DIR="$S3_DEPLOY_DIR/$SLUG/$S3_VERSION_HOSTOS"
+	deploy_build "$S3_DEPLOY_IMAGES_DIR" "false"
+	if [ "$deployTo" == "production" ]; then
+		S3_ACCESS_KEY=${PRODUCTION_S3_ACCESS_KEY}
+		S3_SECRET_KEY=${PRODUCTION_S3_SECRET_KEY}
+		if [ "$RESIN_MANAGED_IMAGE" == "yes" ]; then
+			S3_BUCKET=resin-production-img-cloudformation/images
+		else
+			S3_BUCKET=resin-production-img-cloudformation/resinos
+		fi
+		S3_SYNC_OPTS="$S3_SYNC_OPTS --skip-existing"
+	elif [ "$deployTo" == "staging" ]; then
+		S3_ACCESS_KEY=${STAGING_S3_ACCESS_KEY}
+		S3_SECRET_KEY=${STAGING_S3_SECRET_KEY}
+		if [ "$RESIN_MANAGED_IMAGE" == "yes" ]; then
+			S3_BUCKET=resin-staging-img/images
+		else
+			S3_BUCKET=resin-staging-img/resinos
+		fi
 	else
-		 mv $BUILD_DEPLOY_DIR/$DEPLOY_ARTIFACT $BUILD_DEPLOY_DIR/resin.img
-		(cd $BUILD_DEPLOY_DIR && tar --remove-files --use-compress-program pigz -cvf resin.img.tar.gz resin.img)
+		echo "[ERROR] Refusing to push to deploy than production or master"
```

Comment:
Bit coming from the deployment script. We could improve it in the future. For now this part is imported.